internal/core/src/exec/expression/UnaryExpr.h

```diff
@@ -36,6 +38,26 @@ struct UnaryElementFunc {
             IndexInnerType;
     void
     operator()(const T* src, size_t size, IndexInnerType val, bool* res) {
+        if constexpr (op == proto::plan::OpType::Match) {
+            if constexpr (std::is_same_v<T, std::string_view>) {
+                // translate the pattern match in advance, which avoid computing it every loop.
+                std::regex reg(TranslatePatternMatchToRegex(val));
+                for (int i = 0; i < size; ++i) {
+                    res[i] = std::regex_match(src[i].data(), reg);
```

Comment:
what about performance? should  evaluate it

Reply:
Though this may be very slow, I don't know how to make this faster. Of course, we can compare the different regex library and choose the better.